OMCompiler/Compiler/BackEnd/BackendDAEUtil.mo

```diff
@@ -5352,10 +5353,12 @@ algorithm
         true = expCrefLstHasCref(crexplst,cr1);
         false = Expression.expHasCrefNoPreorDer(e2,cr1);
       then
-        adjacencyRowEnhanced1(rest,e1,e2,vars,globalKnownVars,mark,rowmark,(r,BackendDAE.SOLVABILITY_SOLVED(),{})::inRow,trytosolve,shared);
+        adjacencyRowEnhanced1(rest,e1,e2,vars,globalKnownVars,mark,rowmark,(r,BackendDAE.SOLVABILITY_SOLVED(),{})::inRow,trytosolve,size,shared);
     case(r::rest,_,_,_,_,_,_,_)
       // case: state derivative
       equation
+        // ticket #6806, only allow inverting of function if the sizes match
+        1 = size;
```

Comment:
I prefer using a guard or an if statement instead of throwing an error because `1 != size`. I think it is more easy to understand what happens. But with the comment it's okay for me.

Reply:
Yea i totally agree, but it is programmed this way all through these functions and i did not want to mix up styles, that might be even more confusing.